server/api.go

```diff
@@ -272,3 +272,111 @@ func (d dexAPI) RevokeRefresh(ctx context.Context, req *api.RevokeRefreshReq) (*
 
 	return &api.RevokeRefreshResp{}, nil
 }
+
+func (d dexAPI) CreateConnector(ctx context.Context, req *api.CreateConnectorReq) (*api.CreateConnectorResp, error) {
+	if req.Connector == nil {
+		return nil, errors.New("no connector supplied")
+	}
+	if req.Connector.Id == "" || req.Connector.Type == "" || req.Connector.Name == "" {
+		return nil, errors.New("Connector ID, Type, and Name are mandatory fields")
+	}
+
+	c := storage.Connector{
+		ID:              req.Connector.Id,
+		Type:            req.Connector.Type,
+		Name:            req.Connector.Name,
+		ResourceVersion: "1",
+		Config:          req.Connector.Config,
+	}
+	if err := d.s.CreateConnector(c); err != nil {
```

Comment:
Probably want to make sure the config actually parses too, right? E.g. so people can't provide invalid arguments or JSON.